website/src/introduction.md

```diff
@@ -19,4 +19,19 @@
 
 # Iceberg Rust
 
-Iceberg Rust is a rust implementation for accessing iceberg tables.
+`iceberg-rust` is a Rust implementation for managing Apache Iceberg tables.
+
+## What is Apache Iceberg?
+
+[Apache Iceberg](https://iceberg.apache.org) is a modern, high-performance open table format
+for huge analytic datasets that brings SQL-like tables to processing engines including Spark, Trino, PrestoDB, Flink, Hive and Impala. 
+
+Rather than being a new file type, Iceberg provides a metadata layer that sits on top of formats like Parquet 
+and ORC, ensuring data is organized, accessible, and safe to work with at scale. It introduces features long
+expected in databases such as transactional consistency, schema evolution, and time travel into environments 
+where files are stored directly on systems like Amazon S3 or HDFS.
```

Comment:
I'm not sure if we should mention HDFS. Iceberg does support filesystems, but it's main target storage are object-stores. For example, it does not do any list or rename operations